optimum/habana/transformers/models/mpt/modeling_mpt.py

```diff
@@ -67,9 +67,12 @@ def gaudi_mpt_attention_forward(
             else:
                 key_states = torch.cat([past_key_value[0], key_states], dim=2)
                 value_states = torch.cat([past_key_value[1], value_states], dim=2)
-        past_key_value = [key_states, value_states]
+                past_key_value = [key_states, value_states]
     else:
-        past_key_value = [key_states, value_states]
+        past_key_value = [torch.empty(key_states.shape, dtype=key_states.dtype,device=key_states.device),
+                        torch.empty(key_states.shape, dtype=key_states.dtype,device=key_states.device)]
+        past_key_value[0][:] = key_states[:]
+        past_key_value[1][:] = value_states[:]
```

Comment:
Could you please add your description in PR discussion here as comment for easier code understanding and future references? It's much easier to understand the code if there's comment rather than going thru commits and associated PRs.

Reply:
I changed the `past_key_value` in this PR from tuple type to list to make it compatible with [this](https://github.com/huggingface/optimum-habana/pull/1137) merged PR.